harvester_e2e_tests/integration/test_vm.py

```diff
@@ -0,0 +1,295 @@
+    wait_vm_agent_connected(api_client, second_vm_name, wait_timeout)
+
+    code, first_vm_data = api_client.vms.get_status(first_vm_name)
+    assert 200 == code, (
+        f"Failed to get VM {first_vm_name} with error: {code}, {first_vm_data}"
+    )
+    code, second_vm_data = api_client.vms.get_status(second_vm_name)
+    assert 200 == code, (
+        f"Failed to get VM {second_vm_name} with error: {code}, {second_vm_data}"
+    )
+
+    first_vm_src_host = first_vm_data['status']['nodeName']
+    first_vm_dst_host = ""
+    for node_name in available_node_names:
+        if node_name != first_vm_src_host:
+            first_vm_dst_host = node_name
+    second_vm_src_host = second_vm_data['status']['nodeName']
+    second_vm_dst_host = ""
+    for node_name in available_node_names:
+        if node_name != second_vm_src_host:
+            second_vm_dst_host = node_name
+
+    code, data = api_client.vms.migrate(first_vm_name, first_vm_dst_host)
+    assert code == 204, (
+        f"Can't migrate VM {first_vm_name} from host {first_vm_src_host} to {first_vm_dst_host}"
+    )
+    code, data = api_client.vms.migrate(second_vm_name, second_vm_dst_host)
+    assert code == 204, (
+        f"Can't migrate VM {second_vm_name} from host {second_vm_src_host} to {second_vm_dst_host}"
+    )
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(first_vm_name)
+        if data.get('status', {}).get('migrationState', {}).get('completed', False):
+            assert first_vm_dst_host == data['status']['nodeName'], (
+                f"Failed to migrate VM {first_vm_name} \
+                    from {first_vm_src_host} to {first_vm_dst_host}"
+            )
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"The migration of VM {first_vm_name} is not completed with {wait_timeout} timed out"
+            f"Got error: {code}, {data}"
+        )
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(second_vm_name)
+        if data.get('status', {}).get('migrationState', {}).get('completed', False):
+            assert second_vm_dst_host == data['status']['nodeName'], (
+                f"Failed to migrate VM {second_vm_name} \
+                    from {second_vm_src_host} to {second_vm_dst_host}"
+            )
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"The migration of VM {second_vm_name} is not completed with {wait_timeout} timed out"
+            f"Got error: {code}, {data}"
+        )
+
+    # teardown
+    api_client.vms.delete(first_vm_name)
+    volume_name = ""
+    for volume in first_vm_data['spec']['volumes']:
+        if volume['name'] == 'disk-0':
+            volume_name = volume['persistentVolumeClaim']['claimName']
+    api_client.volumes.delete(volume_name)
```

Comment:
when VM is still terminating, Volumes will not able to delete, we should check whether the VM is removed first.
and we can leverage Spec to remove all volumes like:
```python
for vol in api.vms.Spec.from_dict(data).volumes:
    api.volumes.delete(vol['volume']['persistentVolumeClaim']['claimName'])
```